packages/vite/src/node/plugins/importMetaGlob.ts

```diff
@@ -437,6 +449,11 @@ export async function transformGlobImport(
             let importPath = paths.importPath
             let importQuery = options.query ?? ''
 
+            if (onlyKeys) {
+              objectProps.push(`${JSON.stringify(filePath)}: ''`)
+              return
+            }
+
```

Comment:
If we detect that `import.meta.glob` is wrapped with `Object.keys`, we can skip wrapping the result with `/* #__PURE__ */ Object.assign()` and trim the code a bit more.